lib/cocoapods/installer/analyzer/target_inspector.rb

```diff
@@ -216,16 +216,22 @@ def compute_recommends_frameworks(target_definition, native_targets)
         # @return [String] the targets Swift version or nil
         #
         def compute_swift_version_from_targets(targets)
-          versions = targets.flat_map do |target|
-            target.resolved_build_setting('SWIFT_VERSION').values
-          end.flatten.compact.uniq
-          case versions.count
+          versions_to_targets = targets.inject({}) do |memo, target|
+            versions = target.resolved_build_setting('SWIFT_VERSION').values
+            versions.each do |version|
+              memo[version] = [] if memo[version].nil?
+              memo[version] << target.name unless memo[version].include? target.name
+            end
+            memo
+          end
+
+          case versions_to_targets.count
           when 0
             nil
           when 1
-            versions.first
+            versions_to_targets.keys.first
           else
-            raise Informative, 'There may only be up to 1 unique SWIFT_VERSION per target.'
+            raise Informative, "There may only be up to 1 unique SWIFT_VERSION per target. Found target(s) with multiple Swift versions:\n#{JSON.pretty_generate(versions_to_targets)}"
```

Comment:
Minor issue: Can we have the output formatting just be a newline separated list of <targetname>: Swift <version> ? I think this would be easier to read at first glance.